src/ui/public/chrome/directives/kbn_chrome.js

```diff
@@ -43,6 +43,29 @@ export default function (chrome, internals) {
         // and some local values
         chrome.httpActive = $http.pendingRequests;
         $scope.notifList = require('ui/notify')._notifs;
+
+        // App switcher functionality.
+        function updateAppSwitcher(isOpen) {
+          $scope.appSwitcherButton = {
+            classes: isOpen ? 'app-switcher-link--close' : undefined,
+            title: isOpen ? 'Collapse' : 'Expand',
+            tooltip: isOpen ? 'Collapse side bar' : 'Expand side bar',
+          };
+        }
+
+        $scope.isAppSwitcherOpen = appSwitcherState.isOpen();
+        updateAppSwitcher($scope.isAppSwitcherOpen);
+
+        $scope.toggleAppSwitcher = event => {
+          event.preventDefault();
+          $scope.isAppSwitcherOpen = !$scope.isAppSwitcherOpen;
+          appSwitcherState.setOpen($scope.isAppSwitcherOpen);
+          updateAppSwitcher($scope.isAppSwitcherOpen);
+
+          // Notify visualizations, e.g. the dashboard, that they should re-render.
+          $scope.$root.$broadcast('ready:vis');
```

Comment:
I worry a bit about piggy backing on an existing event. Even if it doesn't have unintended side effects today, it might surprise other devs in the future. If I added a `ready:vis` listener I certainly wouldn't expect it to fire on every sidebar expand/collapse.

What about defining a new event that the chrome can fire, and listening for it in the grid.js file you linked to?